# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerWorker.h
# Commit: bffd879a7547
# Full Hash: bffd879a75476ea682e86082f960e24884d6ca2e
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Implement scheduler.yield r=jjaschke,dom-core,webidl,smaug
#   
#   Spec: https://wicg.github.io/scheduling-apis/#dom-scheduler-yield
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234232
# ==============================================================================

diff -r f441cb5e7e6e -r bffd879a7547 dom/webscheduling/WebTaskSchedulerWorker.h
--- a/dom/webscheduling/WebTaskSchedulerWorker.h	Mon Mar 31 19:49:02 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerWorker.h	Mon Mar 31 19:49:02 2025 +0000
@@ -20,8 +20,7 @@
 
 class WebTaskWorkerRunnable final : public WorkerSameThreadRunnable {
  public:
-  WebTaskWorkerRunnable(WorkerPrivate* aWorkerPrivate,
-                        WebTaskSchedulerWorker* aSchedulerWorker);
+  explicit WebTaskWorkerRunnable(WebTaskSchedulerWorker* aSchedulerWorker);
 
   MOZ_CAN_RUN_SCRIPT_BOUNDARY
   bool WorkerRun(JSContext* aCx, WorkerPrivate* aWorkerPrivate) override;
@@ -42,8 +41,9 @@
  private:
   ~WebTaskSchedulerWorker() = default;
 
-  nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay) override;
-  bool DispatchEventLoopRunnable() override;
+  nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay,
+                                    EventQueuePriority aPriority) override;
+  bool DispatchEventLoopRunnable(EventQueuePriority aPriority) override;
 
   RefPtr<StrongWorkerRef> mWorkerRef;
   bool mWorkerIsShuttingDown{false};